### PR TITLE
[HttpFoundation] Fix caching of session-enabled pages

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #24690
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php [hunk 1]
```diff
@@ -101,6 +102,7 @@ public function __construct(array $options = array(), $handler = null, MetadataB
     {
         $options += array(
             'cache_limiter' => 'private_no_expire',
+            'cache_expire' => 0,
```
Comment (Tobion) [inline src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php]: Can't you then use cache_limiter => nocache directly. The only difference is that it also sets no-store, meaning no conditional requests.

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php]: No-store is too aggressive to me, conditional requests are fine, aren't they?

Comment (Tobion) [reply src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php]: Should be yeah.

**Review by Tobion — APPROVED** [review]

**Comment by Tobion** [comment]
Thank you @nicolas-grekas.

**Pull request merged**